dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPIImpl.java

```diff
@@ -122,31 +95,31 @@ public Field save(Field field, User user) throws DotDataException, DotSecurityEx
         CacheLocator.getContentTypeCache().remove(structure);
         StructureServices.removeStructureFile(structure);
 
-        
-        if(oldField!=null){
-	        if(oldField.indexed() != field.indexed()){
-	          conAPI.refresh(structure);
-	        } else if (field instanceof ConstantField) {
-	        	if(!StringUtils.equals(oldField.values(), field.values()) ){
-		            ContentletServices.removeContentletFile(structure);
-		            ContentletMapServices.removeContentletMapFile(structure);
-		            conAPI.refresh(structure);
-	        	}
-	        }
-        }
-
-        return result;
+      if(oldField!=null){
+          if(oldField.indexed() != field.indexed()){
+              contentletAPI.refresh(structure);
+          } else if (field instanceof ConstantField) {
+              if(!StringUtils.equals(oldField.values(), field.values()) ){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)